client/client.go

```diff
@@ -220,6 +256,40 @@ func (c *Client) processChannelRequests() {
 			c.supervisor.SetState(p)
 			responsePayload = new(agentpb.SetStateResponse)
 
+		case *agentpb.StartActionRequest:
+			var a actions.Action
+			switch p.Name {
```

Comment:
I think no. 

As you can see, it called `ConcurrentRunner` and his main task is to Run an Action concurrently, not creating them by name defined in API. A `Runner` gets an `Action` as a parameter, and it placed in the same package with actions. 

My idea here is that the `action` package is the `application layer`(inside ring), and the client is the `infrastructure layer`(outside ring). So the client can use the application layer code, but the application layer shouldn't depend on API (on `*agentpb.StartActionRequest` for example).

What do you think?

Reply:
yeah, you are right that it shouldn't be part of the action runner, but maybe there should be another layer to be more consistent with the previous code.

Reply:
> but maybe there should be another layer to be more consistent with the previous code
You are right, I agree, it will be better to move action creation somewhere. 

I thought about that. The reason I didn't do this, is because we don't have at least several real actions to realize where we can move that code. Also, I thought about deadlines, so it was a balanced decision.